caikit_nlp/modules/text_embedding/embedding.py

```diff
@@ -281,6 +281,16 @@ def _encode_with_retry(self, *args, **kwargs) -> EmbeddingResultTuple:
             if "batch_size" not in kwargs:
                 kwargs["batch_size"] = BATCH_SIZE
 
+        if isinstance(self.model, SentenceTransformerWithTruncate):
+            return self._with_retry(self.model.encode, *args, **kwargs)
+
+        # Else...
+        # It's possible to init with a model that doesn't have the added kwargs.
+        # E.g. a SentenceTransformer or other transormer model. Remove those kwargs!
+        # This is not the normal use case but at least don't pass invalid kwargs, to encode()
+        # and don't return the unexpected tuple (adding token count).
+        del kwargs["truncate_input_tokens"]
+        del kwargs["return_token_count"]
```

Comment:
might be good to have a small test for this case, in case an update could break it (again)?

Reply:
Yes and I came here to make the PR a draft because this isn't catching keyerrors (noticed while reviewing with Flavia)

Reply:
Yep and done.  Definitely was a silly miss to not check AND not test. :(